azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java

```diff
@@ -236,11 +246,19 @@ private void uploadFlowFilesRecursively(final File projectDir, final Project pro
   }
 
   private void cleanUpProjectOldInstallations(final Project project)
-      throws ProjectManagerException{
+      throws ProjectManagerException, ExecutorManagerException {
     log.info("Cleaning up old install files older than "
         + (project.getVersion() - this.projectVersionRetention));
+    // TODO fetch high level metadata without DAG object
+    Map<Integer, Pair<ExecutionReference, ExecutableFlow>> activeFlows = this.executorLoader
```

Comment:
What about queued flows? As you mentioned, there might be a delay in dispatching to executors and the executions are queued on the web server. Should we exclude the project versions from being deleted for those queued flows as well? If that's the case, you can use `this.executorLoader.fetchUnfinishedFlows` which will include the queued flows.